src/dials/command_line/dials_import.py

```diff
@@ -361,17 +362,27 @@ def __call__(self, imageset):
             for j in range(len(imageset)):
                 imageset.set_scan(None, j)
                 imageset.set_goniometer(None, j)
+
+        beam = imageset.get_beam()
+        detector = imageset.get_detector()
+        goniometer = imageset.get_goniometer()
+        scan = imageset.get_scan()
+
         if isinstance(imageset, ImageSequence):
-            beam = BeamFactory.from_phil(self.params.geometry, imageset.get_beam())
-            detector = DetectorFactory.from_phil(
-                self.params.geometry, imageset.get_detector(), beam
-            )
-            goniometer = GoniometerFactory.from_phil(
-                self.params.geometry, imageset.get_goniometer()
-            )
-            scan = ScanFactory.from_phil(
-                self.params.geometry, deepcopy(imageset.get_scan())
-            )
+            if beam and beam not in self.touched:
```

Comment:
I am not convinced that the `if beam and beam not in self.touched` logic here is consistent with the `beam = BeamFactory()` since every new beam will be a beam? To be precise, I think I would like a unit test which demonstrates that the logic works as advertised.

Reply:
This is more for the `goniometer` and `scan` models, which could be `None`, in which case I thought it should not try to override the geometry. No sane imageset would give `None` for `get_beam()` or `get_detector()`, but I figured the logic could just as well be consistent for each model. I'll think about a suitable test,,,